src/foundation/src/PDFsharp/src/PdfSharp/Pdf/PdfDictionary.cs

```diff
@@ -45,10 +45,38 @@ public class PdfDictionary : PdfObject, IEnumerable<KeyValuePair<string, PdfItem
         // Reference: 3.2.6  Dictionary Objects / Page 59
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="PdfDictionary"/> class.
+        /// Gets a value that determines whether the object was modified after loading.
         /// </summary>
-        public PdfDictionary()
-        { }
+        internal bool IsModified { get; private set; }
+
+        /// <summary>
+        /// Sets the modified-status of this object
+        /// </summary>
+        /// <param name="modified"></param>
+        internal void SetModified(bool modified)
+        {
+            if (Owner == null || !Owner.IsAppending || !Owner.IrefTable.FullyLoaded)
```

Comment:
I believe this null check should not be here otherwise the IrefTable is not updated

Reply:
Not sure if i get your point here, you mean the `if (Owner == null` - check ?
The code would crash without it (on the `Owner.IsAppending` statement)

Reply:
Yeah it was added by me

Reply:
To me it feels that we should ensure owner is never null rather than null check here